smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/RuntimeConfigGenerator.java

```diff
@@ -269,6 +269,105 @@ private void generateHttpAuthSchemeConfig(
         });
     }
 
+    private static class HttpAuthSchemeTarget {
+        public HttpAuthScheme httpAuthScheme;
+        public Consumer<TypeScriptWriter> identityProvider;
+        public Consumer<TypeScriptWriter> signer;
+
+        HttpAuthSchemeTarget(
+            HttpAuthScheme httpAuthScheme,
+            Consumer<TypeScriptWriter> identityProvider,
+            Consumer<TypeScriptWriter> signer
+        ) {
+            this.httpAuthScheme = httpAuthScheme;
+            this.identityProvider = identityProvider;
+            this.signer = signer;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (!(other instanceof HttpAuthSchemeTarget)) {
+                return false;
+            }
+            HttpAuthSchemeTarget o = (HttpAuthSchemeTarget) other;
+            return httpAuthScheme.equals(o.httpAuthScheme)
+                && identityProvider.equals(o.identityProvider)
+                && signer.equals(o.signer);
+        }
+
+        @Override
+        public int hashCode() {
+            return super.hashCode();
+        }
```

Comment:
is this the same if you leave this override out?

Reply:
Yes, but `Checkstyle` will report a error violation:

```
Definition of 'equals()' without corresponding definition of 'hashCode()'.
```